app/webpacker/packs/cookie-banner.js

```diff
@@ -0,0 +1,44 @@
+const cookieBannerEl = document.querySelector(".js-cookie-banner");
+
+if (cookieBannerEl) {
+  const cookieFormEl = document.querySelector(".js-cookie-form");
+
+  cookieFormEl.addEventListener("click", (e) => {
+    if (e.target.tagName !== "BUTTON") {
+      return;
+    }
+
+    fetch(cookieFormEl.action, {
+      method: "PUT",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        cookies_form: { analytics_consent: e.target.value },
+      }),
+    })
+      .then((res) => res.json())
+      .then(({ message }) => {
+        const messageEl = cookieBannerEl.querySelector(".js-cookie-message");
+        messageEl.textContent = message;
+
+        Array.from(cookieBannerEl.children).forEach((childEl) => {
+          if (childEl.hasAttribute("hidden")) {
+            childEl.removeAttribute("hidden");
+          } else {
+            childEl.setAttribute("hidden", "");
+          }
+        });
```

Comment:
Can we have ids on the banner elements, like `site-uses-cookies-message`, `cookies-set-message` and hide / unhide them by hand? Those are the only two elements, right? I think it would make it a lot easier to figure out what to expect.